mockbroker.go

```diff
@@ -50,18 +51,22 @@ type MockBroker struct {
 	closing      chan none
 	stopper      chan none
 	expectations chan encoder
+	done         sync.WaitGroup
 	listener     net.Listener
 	t            TestReporter
 	latency      time.Duration
 	handler      requestHandlerFunc
-	history      []RequestResponse
+	origHandler  bool
+	history      []*RequestResponse
```

Comment:
It is because of the way I am updating the `ResponseSize` field, I first add an entry to the `history` field with the default value 0 and keep a reference to that `RequestResponse` before potentially updating it at the end of the handling.